kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -118,30 +129,56 @@ public CloseableIterator<FilteredColumnarBatch> getScanFiles(
     boolean hasDataSkippingFilter = dataSkippingFilter.isPresent();
     boolean shouldReadStats = hasDataSkippingFilter || includeStats;
 
-    // Get active AddFiles via log replay
-    // If there is a partition predicate, construct a predicate to prune checkpoint files
-    // while constructing the table state.
-    CloseableIterator<FilteredColumnarBatch> scanFileIter =
-        logReplay.getAddFilesAsColumnarBatches(
-            engine,
-            shouldReadStats,
-            getPartitionsFilters()
-                .map(
-                    predicate ->
-                        rewritePartitionPredicateOnCheckpointFileSchema(
-                            predicate, partitionColToStructFieldMap.get())));
-
-    // Apply partition pruning
-    scanFileIter = applyPartitionPruning(engine, scanFileIter);
-
-    // Apply data skipping
-    if (hasDataSkippingFilter) {
-      // there was a usable data skipping filter --> apply data skipping
-      scanFileIter = applyDataSkipping(engine, scanFileIter, dataSkippingFilter.get());
-    }
+    Timer.Timed planningDuration = scanMetrics.totalPlanningTimer.start();
+    ScanReportReporter reportReporter =
```

Comment:
so this is like a ... context wrapper? you can invoke the reporter with .reportError or .reportSuccess, and it goes and executes this code, like stopping the planning duration and pushing the scanReport to all the engine's metrics reporters?

makes sense!

maybe a comment here that like we are just building this context wrapper? ... it itself isn't actually invoking anything yet. (just cuz it took me > 30 seconds to look at this and figure out what was going on)

Reply:
sure will add a comment